src/App.tsx

```diff
@@ -1,61 +1,97 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('0');
+  const [todolist, setTodolist] = useState(todosFromServer);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const toDoData = todolist.map(todo => {
+    return {
+      ...todo,
+      userId: getUser(todo.userId),
+    };
+  });
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        onSubmit={(event) => {
+          event.preventDefault();
+          setTitleError(!title.trim());
+          setUserError(user === '0');
+          if (title.trim() !== '' && user !== '0') {
+            setTodolist([
+              ...todolist,
+              {
+                id: Math.max(...todolist.map(todo => todo.id)) + 1,
+                title: title.toLowerCase(),
+                completed: false,
+                userId: Number(user),
+              },
+            ]);
+            setTitle('');
+            setUser('0');
+          }
+        }}
```

Comment:
it would be better to create separate function `onSubmit` and move the code there